doc/source_typing_3.tex

```diff
@@ -0,0 +1,46 @@
+
+\section*{Typing}
+
+Expressions evaluate to numbers, boolean values, strings, function objects and arrays. 
+
+Only function values can be applied using the syntax:
+
+\begin{eqnarray*}
+ \textit{expression}    
+                                   & ::=   &  \textit{name}
+                                               \texttt{\textbf{(}}\  \textit{expressions} \
+                                               \texttt{\textbf{)}}\\ 
+\end{eqnarray*}
+
+The following table specifies what arguments Source's operators
+take and what results they return.
+
+\begin{center}
+\begin{tabular}{c|c|c|c}
+operator & argument 1 & argument 2 & result\\ \hline
+\texttt{\textbf{+}} & number   & number     & number\\
+\texttt{\textbf{+}} & string   & any        & string\\
+\texttt{\textbf{+}} & any      & string     & string\\
+\texttt{\textbf{-}} & number   & number     & number\\
+\texttt{\textbf{*}} & number   & number     & number\\
+\texttt{\textbf{/}} & number   & number     & number\\
+\texttt{\textbf{\%}} & number   & number     & number\\
+\texttt{\textbf{===}} & any   & any     & bool\\
+\texttt{\textbf{!==}} & any   & any     & bool\\
+\texttt{\textbf{>}} & number   & number     & bool\\
+\texttt{\textbf{>}} & string   & string     & bool\\
+\texttt{\textbf{<}} & number   & number     & bool\\
+\texttt{\textbf{<}} & string   & string     & bool\\
+\texttt{\textbf{>=}} & number   & number     & bool\\
+\texttt{\textbf{>=}} & string   & string     & bool\\
+\texttt{\textbf{<=}}    & number   & number     & bool\\
+\texttt{\textbf{<=}} & string   & string     & bool\\
+\texttt{\textbf{\&\&}} & bool & bool & bool\\
+\texttt{\textbf{||}}   & bool & bool & bool\\
```

Comment:
For `&&` and `||` are the specs going to change to bool op any = any?

Reply:
Yes. We don't make an explicit claim about this, but currently the wording suggests that (or at least leaves open the question whether) there are runtime tests for this. As it stands, the translation to conditional expressions would prevent such a runtime test.